src/models/mnist_module.py

```diff
@@ -59,6 +60,16 @@ def on_train_start(self):
         # so we need to make sure val_acc_best doesn't store accuracy from these checks
         self.val_acc_best.reset()
 
+        # although val_loss and val_acc are automatically reset at end of sanity check (since they're fed as
+        # torchmetrics.Metric objects to self.log in validation_step with on_epoch=True), reset them anyways
+        self.val_loss.reset()
+        self.val_acc.reset()
```

Comment:
I'm confused by the reasoning here. Is resetting all validation metrics necessary before the progress bar fix later on?

Reply:
Athough we could encourage resetting all validiation metrics at the start of the training like this:
```python
# by default lightning executes validation step sanity checks before training starts,
# so we need to make sure validation metrics don't store results from these checks
self.val_acc_best.reset()
self.val_loss.reset()
self.val_acc.reset()
```

(just in case it will save someone trouble when doing something unconventional with validation metrics)

Reply:
Yes, the suggestion above makes more sense. It's what I originally would have put so as to encourage resetting of all validation metrics, even though val_loss and val_acc are in fact resetted.

To answer your Q: no, it's not necessary to reset all validation metrics for the progress bar fix. However, instead of explaining why that's the case to the user—which is long-winded and nuanced (as you saw in my code comment above)—it's probably better to just encourage explicitly resetting any validation metrics that were logged in either `validation_step()` or `validation_epoch_end()` callbacks.

Reply:
Regarding the reset of all validation metrics when training starts - thank you for this suggestion, I've made appriopriate PR here: https://github.com/ashleve/lightning-hydra-template/pull/540